services/src/main/java/org/exoplatform/addons/gamification/storage/dao/RuleDAO.java

```diff
@@ -95,6 +96,18 @@ public RuleEntity findRuleByEventAndDomain(String event, String domain) throws P
 
     }
 
+    public List<RuleEntity> getActiveRules() throws PersistenceException {
+
+        TypedQuery<RuleEntity> query = getEntityManager().createNamedQuery("Rule.getEnabledRules", RuleEntity.class)
+                .setParameter("isEnabled",true);
+        try {
+            return query.getResultList();
+        } catch (NoResultException e) {
+            return Collections.emptyList();
+        }
+
+    }
+
```

Comment:
```suggestion
    public List<RuleEntity> getActiveRules() {

        
        try {
            TypedQuery<RuleEntity> query = getEntityManager().createNamedQuery("Rule.getEnabledRules", RuleEntity.class)
                .setParameter("isEnabled",true);
            return query.getResultList();
        } catch (PersistenceExcetion e) {
            LOG.error("blabla to complete",e);
            return Collections.emptyList();
        } catch (NoResultException e) {
            LOG.error("blabla to complete",e);
            return Collections.emptyList();
        }

    }

```

Reply:
according to the compilation error, NoResultException should be catched the first and the only one, if i catch an other exception the compiler will ignore that catch scope.
**'catch' branch identical to 'NoResultException' branch** this warning message appears when use a second catch scope in this case.

Reply:
did you try this syntax 
try {
     // stuff
} catch (Exception1 | Exception2 ex) {
     // Handle both exceptions
}